test/run_tests/smoke_test/threads_for_profiling.rb

```diff
@@ -0,0 +1,255 @@
+    puts "#{indent}Label:   #{trace["Label"]}"
+    puts "#{indent}Layer:   #{trace["Layer"]}"
+
+    more_keys.each { |key| puts "#{indent}#{key}:   #{trace[key]}" if trace[key] }
+
+    indent = indent[0...-2] if trace["Label"] == "exit"
+  end
+  puts "\n"
+end
+
+
+def massive_fun(temps)
+  aa = Array.new
+  4.times do
+  5.times do
+    5.times do
+      4.times do
+        temps.sort.reverse
+        aa << "it is too #{temps.shuffle!.first}"
+        aa.reverse.size
+        # puts "............................... #{aa.last} ......................."
+        aa.delete_if do |a|
+          a =~ /[l|r]/
+        end
+        # sleep 0.01
+      end
+    end
+  end
+  # sleep 0.01
+  end
+end
+
+class Hola
+  def self.my_fun
+    1.times do
+      sum_prime = 0
+      Prime.each(10000) do |prime|
+        sum_prime += prime
+        sum_prime/2
+      end
+    end
+  end
+end
+
+class Kids
+  def giggle
+    temps = ['colder', 'warmer', 'chilly','hot']
+    # ts = AppOpticsAPM::TransactionSettings.new
+    # 100_000.times do
+    #   ts.to_s
+    # end
+    # massive_fun(temps)
+  end
+end
+
+class HaHaHa
+  def laughing
+    temps = ['colder', 'warmer', 'chilly','hot']
+    massive_fun(temps)
+  end
+end
+
+# AppOpticsAPM::SDK.trace_method(HaHaHa, :laughing)
+# AppOpticsAPM::SDK.trace_method(Kids, :giggle) if defined? AppOpticsAPM
+# AppOpticsAPM::SDK.trace_method(Hola, :my_fun) if defined? AppOpticsAPM
+
+AppOpticsAPM::Config.profiling = :enabled if defined? AppOpticsAPM
+AppOpticsAPM::Config[:profiling_interval] = 5 if defined? AppOpticsAPM
+
+unless AppOpticsAPM::SDK.appoptics_ready?(10_000)
+  puts "aborting!!! Agent not ready after 10 seconds"
+  exit false
+end if defined? AppOpticsAPM
+
+
+class A
+  def initialize
+    pow
+    self.class.newobj
+    math
+    # Array(1..100).reverse
+    # File.open("log.txt", "w") { |f| f.write "#{Time.now} - timestamped!!!\n" }
+
+  end
+
+  def pow
+    2 ** 100
+  end
+
+  def self.newobj
+    Object.new
+    Object.new
+  end
+
+  def A.math
+    2.times do
+      # print "."
+      2 + 3 * 4 ^ 5 / 6
+    end
+  end
+end
+
+class Stuff
+  def Stuff.do_stuff
+    100.times { a.math }
+  end
+end
+  # threads = []
+
+  # puts "Main tid: #{AppOpticsAPM::CProfiler.get_tid}"
+  # AppOpticsAPM::SDK.start_trace("main_thread") do
+  #   AppOpticsAPM::Profiling.run do
+  # 3.times do |i|
+  #   threads << Thread.new do
+  #     sleep 0.1
+  # i = 0
+  # 10.times do
+  # AppOpticsAPM::SDK.start_trace("thread-#{i}") do
+    # AppOpticsAPM::Profiling.run do
+      # tid = AppOpticsAPM::CProfiler.get_tid
+      # puts "thread tid: #{tid}, tracing? #{AppOpticsAPM.tracing?}"
+
+      # AppOpticsAPM::SDK.trace(:boo) do
+      #   50.times do
+      # File.open("foo_#{i}.txt", 'w') { |f| f.write(Time.now) }
+      # A.new
+      # HaHaHa.new.laughing
+      # i+i
+      # sleep 1
+      # end
+    # end
+  # end
+  # end
+  # sleep 0.2
+  # end
+# end
+  # pid = fork do
+    threads = []
```

Comment:
Is this intentinally left uncommented?

Reply:
sorry, this is a script I run manually depending on what I'm focussing on, I should add a comment somewhere